src/backend/executor/nodeDML.c

```diff
@@ -45,158 +65,199 @@ ExecDMLExplainEnd(PlanState *planstate, struct StringInfoData *buf)
 TupleTableSlot*
 ExecDML(DMLState *node)
 {
+	for (;;)
+	{
+		PlanState *outerNode = outerPlanState(node);
+		DML *plannode = (DML *) node->ps.plan;
 
-	PlanState *outerNode = outerPlanState(node);
-	DML *plannode = (DML *) node->ps.plan;
+		Assert(outerNode != NULL);
 
-	Assert(outerNode != NULL);
+		/* Temporary restore result tuple slot for use in next projection */
```

Comment:
Maybe
```suggestion
		/* Temporary store result tuple slot for use in next projection */
```
as we are storing old values for future use?

Reply:
I meant here we are restoring resultTuple which will be used just in ExecProject on 112 line
On 118 line we are restoring resultTuple which will be used returning projection